src/backend/distributed/sql/udfs/drop_old_time_partitions/latest.sql

```diff
@@ -0,0 +1,44 @@
+-- Heavily inspired by the procedure alter_old_partitions_set_access_method
+CREATE OR REPLACE PROCEDURE pg_catalog.drop_old_time_partitions(
+	parent_table_name regclass,
+	older_than timestamptz)
+LANGUAGE plpgsql
+AS $$
+DECLARE
+    r record;
+	schema_name_text text;
+BEGIN
+	/* first check whether we can convert all the to_value's to timestamptz */
+	BEGIN
+		PERFORM
+		FROM pg_catalog.time_partitions
+		WHERE parent_table = parent_table_name
+		AND to_value IS NOT NULL
+		AND to_value::timestamptz <= older_than;
+	EXCEPTION WHEN invalid_datetime_format THEN
+		RAISE 'partition column of % cannot be cast to a timestamptz', parent_table_name;
+	END;
+
+	SELECT nspname
+    INTO schema_name_text
+    FROM pg_class JOIN pg_namespace ON pg_class.relnamespace = pg_namespace.oid 
+    WHERE pg_class.oid = parent_table_name::oid;
+
+	/* now drop the partitions in separate transactions */
+    FOR r IN
+		SELECT partition, from_value, to_value
+		FROM pg_catalog.time_partitions
+		WHERE parent_table = parent_table_name
+		AND to_value IS NOT NULL
+		AND to_value::timestamptz <= older_than
+		ORDER BY to_value::timestamptz
+    LOOP
+        RAISE NOTICE 'dropping % with start time % and end time %', r.partition, r.from_value, r.to_value;
+        EXECUTE format('DROP TABLE %I.%I', schema_name_text, r.partition);
```

Comment:
partition is a regclass, so it might include a schema name when converting to string. A join with pg_class is required.

alter_old_time_partitions_set_access_method can avoid this because it passes the OID/regclass directly to pg_catalog.alter_table_set_access_method